File: backend/src/providers/menuItemImageProvider.ts
import { prisma } from '../lib/prisma.js';

export const menuItemImageProvider = {
  async createMany(
    menuItemId: number,
    images: Array<{ image_url: string; is_primary?: boolean; display_order?: number }>,
  ) {
    return prisma.menu_item_images.createMany({
      data: images.map((img) => ({
        menu_item_id: menuItemId,
        image_url: img.image_url,
        is_primary: img.is_primary ?? false,
        display_order: img.display_order ?? 0,
      })),
    });
  },

  async findById(id: number) {
    return prisma.menu_item_images.findFirst({ where: { id } });
  },

  async findByMenuItemId(menuItemId: number) {
    return prisma.menu_item_images.findMany({
      where: { menu_item_id: menuItemId },
      orderBy: [{ is_primary: 'desc' }, { display_order: 'asc' }],
    });
  },

  async delete(id: number) {
    return prisma.menu_item_images.delete({ where: { id } });
  },
};

File: backend/app/providers/menu_item_image_provider.py
from sqlalchemy.ext.asyncio import AsyncSession
from sqlalchemy import select, update, delete
from typing import Optional, List
from app.models.menu_item_image import MenuItemImage

class MenuItemImageProvider:
    async def find_by_menu_item(self, db: AsyncSession, menu_item_id: int) -> List[MenuItemImage]:
        result = await db.execute(
            select(MenuItemImage)
            .where(MenuItemImage.menu_item_id == menu_item_id)
            .order_by(MenuItemImage.is_primary.desc(), MenuItemImage.display_order.asc())
        )
        return list(result.scalars().all())

    async def create(self, db: AsyncSession, menu_item_id: int, image_url: str, is_primary: bool = False, display_order: int = 0) -> MenuItemImage:
        img = MenuItemImage(menu_item_id=menu_item_id, image_url=image_url, is_primary=is_primary, display_order=display_order)
        db.add(img)
        await db.commit()
        await db.refresh(img)
        return img

    async def set_primary(self, db: AsyncSession, menu_item_id: int, image_id: int) -> None:
        await db.execute(update(MenuItemImage).where(MenuItemImage.menu_item_id == menu_item_id).values(is_primary=False))
        await db.execute(update(MenuItemImage).where(MenuItemImage.id == image_id).values(is_primary=True))
        await db.commit()

    async def delete(self, db: AsyncSession, image_id: int) -> None:
        await db.execute(delete(MenuItemImage).where(MenuItemImage.id == image_id))
        await db.commit()

menu_item_image_provider = MenuItemImageProvider()
